File: delivery-service/db/src/main/java/org/delivery/db/storeUser/StoreUserRepository.java
package org.delivery.db.storeUser;

import org.delivery.db.storeUser.enums.StoreUserStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

public interface StoreUserRepository extends JpaRepository<StoreUserEntity, Long> {

    Optional<StoreUserEntity> findFirstByEmailAndStatusOrderByIdDesc(String email, StoreUserStatus status);

}

File: delivery-service/db/src/main/kotlin/org/delivery/db/storeUser/StoreUserRepository.kt
package org.delivery.db.storeUser

import org.delivery.db.storeUser.enums.StoreUserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface StoreUserRepository : JpaRepository<StoreUserEntity, Long> {
    fun findFirstByEmailAndStatusOrderByIdDesc(email: String?, status: StoreUserStatus?): StoreUserEntity?
}
